## Stock Sentiment Analysis using News Headlines

In [1]:
import pandas as pd

pandas is a powerful open-source data manipulation and analysis library in Python. It provides data structures and functions to efficiently work with structured data, such as tabular data, time series, and more.

By importing pandas as pd, you can refer to its functions and objects using the pd prefix. This is a common convention used by developers to make code more concise and readable.

In [3]:
df=pd.read_csv('/kaggle/input/stock-sentiment-analysis-data/stock_senti_analysis.csv', encoding = "ISO-8859-1")

In [4]:
df.head()

Date  Label                                               Top1  \
0  2000-01-03      0  A 'hindrance to operations': extracts from the...   
1  2000-01-04      0                                          Scorecard   
2  2000-01-05      0                  Coventry caught on counter by Flo   
3  2000-01-06      1                      Pilgrim knows how to progress   
4  2000-01-07      1                               Hitches and Horlocks   

                                 Top2  \
0                           Scorecard   
1                 The best lake scene   
2  United's rivals on the road to Rio   
3                 Thatcher facing ban   
4      Beckham off but United survive   

                                            Top3  \
0                Hughes' instant hit buoys Blues   
1                  Leader: German sleaze inquiry   
2  Thatcher issues defence before trial by video   
3        McIlroy calls for Irish fighting spirit   
4                        Breast cancer screening   

                                            Top4  \
0       Jack gets his skates on at ice-cold Alex   
1                                  Cheerio, boyo   
2  Police help Smith lay down the law at Everton   
3                Leicester bin stadium blueprint   
4                                    Alan Parker   

                                          Top5  \
0       Chaos as Maracana builds up for United   
1                     The main recommendations   
2  Tale of Trautmann bears two more retellings   
3               United braced for Mexican wave   
4      Guardian readers: are you all whingers?   

                                                Top6  \
0  Depleted Leicester prevail as Elliott spoils E...   
1                             Has Cubie killed fees?   
2                                England on the rack   
3  Auntie back in fashion, even if the dress look...   
4                                   Hollywood Beyond   

                                              Top7  \
0                 Hungry Spurs sense rich pickings   
1                           Has Cubie killed fees?   
2  Pakistan retaliate with call for video of Walsh   
3                    Shoaib appeal goes to the top   
4                               Ashes and diamonds   

                                                Top8  \
0                  Gunners so wide of an easy target   
1                             Has Cubie killed fees?   
2               Cullinan continues his Cape monopoly   
3  Hussain hurt by 'shambles' but lays blame on e...   
4                   Whingers - a formidable minority   

               ...               \
0              ...                
1              ...                
2              ...                
3              ...                
4              ...                

                                               Top16  \
0           Flintoff injury piles on woe for England   
1                               On the critical list   
2                        South Melbourne (Australia)   
3  Putin admits Yeltsin quit to give him a head s...   
4                             Most everywhere:  UDIs   

                                               Top17  \
0  Hunters threaten Jospin with new battle of the...   
1                          The timing of their lives   
2                                    Necaxa (Mexico)   
3         BBC worst hit as digital TV begins to bite   
4                       Most wanted:  Chloe lunettes   

                                            Top18  \
0             Kohl's successor drawn into scandal   
1                                     Dear doctor   
2                             Real Madrid (Spain)   
3                     How much can you pay for...   
4  Return of the cane 'completely off the agenda'   

                                               Top19  \
0               The difference between men and women   
1  Irish court halts IRA man's extradition to Nor...   
2            

In [6]:
train = df[df['Date'] < '20150101']
test = df[df['Date'] > '20141231']

Overall, this code is separating the data in the df DataFrame into two subsets based on the dates. The train DataFrame contains the data with dates before '20150101', while the test DataFrame contains the data with dates after '20141231'. This type of separation is commonly done in machine learning or time series analysis to split the data into a training set and a testing (or validation) set for model development and evaluation.
spliting testing and training data sets

In [7]:
# Removing punctuations
data=train.iloc[:,2:27]
data.replace("[^a-zA-Z]"," ",regex=True, inplace=True)

# Renaming column names for ease of access
list1= [i for i in range(25)]
new_Index=[str(i) for i in list1]
data.columns= new_Index
data.head(5)



0  \
0  A  hindrance to operations   extracts from the...   
1                                          Scorecard   
2                  Coventry caught on counter by Flo   
3                      Pilgrim knows how to progress   
4                               Hitches and Horlocks   

                                    1  \
0                           Scorecard   
1                 The best lake scene   
2  United s rivals on the road to Rio   
3                 Thatcher facing ban   
4      Beckham off but United survive   

                                               2  \
0                Hughes  instant hit buoys Blues   
1                  Leader  German sleaze inquiry   
2  Thatcher issues defence before trial by video   
3        McIlroy calls for Irish fighting spirit   
4                        Breast cancer screening   

                                               3  \
0       Jack gets his skates on at ice cold Alex   
1                                  Cheerio  boyo   
2  Police help Smith lay down the law at Everton   
3                Leicester bin stadium blueprint   
4                                    Alan Parker   

                                             4  \
0       Chaos as Maracana builds up for United   
1                     The main recommendations   
2  Tale of Trautmann bears two more retellings   
3               United braced for Mexican wave   
4      Guardian readers  are you all whingers    

                                                   5  \
0  Depleted Leicester prevail as Elliott spoils E...   
1                             Has Cubie killed fees    
2                                England on the rack   
3  Auntie back in fashion  even if the dress look...   
4                                   Hollywood Beyond   

                                                 6  \
0                 Hungry Spurs sense rich pickings   
1                           Has Cubie killed fees    
2  Pakistan retaliate with call for video of Walsh   
3                    Shoaib appeal goes to the top   
4                               Ashes and diamonds   

                                                   7  \
0                  Gunners so wide of an easy target   
1                             Has Cubie killed fees    
2               Cullinan continues his Cape monopoly   
3  Hussain hurt by  shambles  but lays blame on e...   
4                   Whingers   a formidable minority   

                                                   8  \
0      Derby raise a glass to Strupar s debut double   
1  Hopkins  furious  at Foster s lack of Hannibal...   
2             McGrath puts India out of their misery   
3                      England s decade of disasters   
4                             Alan Parker   part two   

                                          9              ...               \
0  Southgate strikes  Leeds pay the penalty              ...                
1                    Has Cubie killed fees               ...                
2            Blair Witch bandwagon rolls on              ...                
3      Revenge is sweet for jubilant Cronje              ...                
4                 Thuggery  Toxins and Ties              ...                

                                                  15  \
0           Flintoff injury piles on woe for England   
1                               On the critical list   
2                        South Melbourne  Australia    
3  Putin admits Yeltsin quit to give him a head s...   
4                             Most everywhere   UDIs   

                                                  16  \
0  Hunters threaten Jospin with new battle of the...   
1                          The timing of their lives   
2                                    Necaxa  Mexico    
3         BBC worst hit as digital TV begins to bite   
4                       Most wanted   Chloe lunettes   

                                               17  \
0             

Overall, this code is separating the data in the df DataFrame into two subsets based on the dates. The train DataFrame contains the data with dates before '20150101', while the test DataFrame contains the data with dates after '20141231'. This type of separation is commonly done in machine learning or time series analysis to split the data into a training set and a testing (or validation) set for model development and evaluation.This line uses the replace() method to remove punctuations from the data DataFrame. It replaces any character that is not a letter (denoted by the regular expression pattern [^a-zA-Z]) with a space. The regex=True parameter indicates that the pattern is a regular expression. The inplace=True parameter modifies the data DataFrame in place, without creating a new DataFrame.

This step is often performed to clean text data by removing non-alphabetic characters, such as punctuation marks or special characters, to prepare the text for further analysis or natural language processing tasks.These lines rename the column names of the data DataFrame for ease of access. It creates a list list1 containing integers from 0 to 24. Then, it creates a new list new_Index by converting each integer in list1 to a string using a list comprehension.

Finally, it assigns the new_Index list as the new column names of the data DataFrame using the columns attribute.

This step is optional but can be useful when you want to have more meaningful or descriptive column names instead of the default column names.This line displays the first five rows of the data DataFrame using the head() method. It helps to quickly inspect the resulting DataFrame and verify the changes made.

Overall, the code snippet you provided is preprocessing the train DataFrame by selecting specific columns, removing punctuations, and renaming the column names for convenience. The resulting DataFrame is stored in the data variable, and the first five rows are displayed.

In [8]:
# Convertng headlines to lower case
for index in new_Index:
    data[index]=data[index].str.lower()
data.head(1)

0          1  \
0  a  hindrance to operations   extracts from the...  scorecard   

                                 2                                         3  \
0  hughes  instant hit buoys blues  jack gets his skates on at ice cold alex   

                                        4  \
0  chaos as maracana builds up for united   

                                                   5  \
0  depleted leicester prevail as elliott spoils e...   

                                  6                                  7  \
0  hungry spurs sense rich pickings  gunners so wide of an easy target   

                                               8  \
0  derby raise a glass to strupar s debut double   

                                          9         ...          \
0  southgate strikes  leeds pay the penalty         ...           

                                         15  \
0  flintoff injury piles on woe for england   

                                                  16  \
0  hunters threaten jospin with new battle of the...   

                                    17                                    18  \
0  kohl s successor drawn into scandal  the difference between men and women   

                                    19  \
0  sara denver  nurse turned solicitor   

                                               20  \
0  diana s landmine crusade put tories in a panic   

                                                  21                22  \
0  yeltsin s resignation caught opposition flat f...  russian roulette   

         23                  24  
0  sold out  recovering a title  

[1 rows x 25 columns]

The code  converts the headlines in the data DataFrame to lowercase. This loop iterates over each column index in the new_Index list, which represents the column names of the data DataFrame. For each column, it uses the .str.lower() method to convert the values in that column to lowercase.

The .str accessor provides a set of string methods that can be applied to each element of a column. In this case, .lower() is used to convert the strings to lowercase.

The loop iterates over all the columns, and the corresponding columns are modified in-place with the lowercase values.data.head(1)
This line displays the first row of the modified data DataFrame after converting the headlines to lowercase.

By converting the headlines to lowercase, you ensure that the text is normalized and consistent, which can be helpful for tasks like text matching, analysis, or natural language processing that are case-insensitive.

Note that the data DataFrame has undergone several preprocessing steps as shown in the previous code snippets.

In [14]:
' '.join(str(x) for x in data.iloc[1,0:25])

'scorecard the best lake scene leader  german sleaze inquiry cheerio  boyo the main recommendations has cubie killed fees  has cubie killed fees  has cubie killed fees  hopkins  furious  at foster s lack of hannibal appetite has cubie killed fees  a tale of two tails i say what i like and i like what i say elbows  eyes and nipples task force to assess risk of asteroid collision how i found myself at last on the critical list the timing of their lives dear doctor irish court halts ira man s extradition to northern ireland burundi peace initiative fades after rebels reject mandela as mediator pe points the way forward to the ecb campaigners keep up pressure on nazi war crimes suspect jane ratcliffe yet more things you wouldn t know without the movies millennium bug fails to bite'

The code snippet ' '.join(str(x) for x in data.iloc[1,0:25]) performs a concatenation of the values in the second row (index 1) of the data DataFrame, specifically columns 0 to 24 (excluding column 25). The data.iloc[1, 0:25] expression selects the second row (index 1) of the data DataFrame and the columns 0 to 24 (excluding column 25). This creates a pandas Series object containing the values from the selected cells.

The code then uses a generator expression (str(x) for x in data.iloc[1, 0:25]) to convert each value in the Series to a string.

Finally, ' '.join(...) is used to join the string representation of each value with a space in between. The resulting string is the concatenation of all the values in the selected row and columns, separated by a space.

This operation can be useful if you want to concatenate the text values from multiple columns into a single string, which can be further processed or analyzed as a unified text entity.

If you run the provided code snippet, it will return the concatenated string of the values from the second row and columns 0 to 24 in the data DataFrame.

In [15]:
headlines = []
for row in range(0,len(data.index)):
    headlines.append(' '.join(str(x) for x in data.iloc[row,0:25]))

The code snippet  is iterating over each row of the data DataFrame and concatenating the values from columns 0 to 24 (excluding column 25) of each row into a single string. The resulting strings are then appended to a list called headlines.In this loop, row iterates over the range of row indices from 0 to the length of the index of the data DataFrame.

For each iteration, ' '.join(str(x) for x in data.iloc[row, 0:25]) concatenates the values in columns 0 to 24 (excluding column 25) of the current row into a single string. This is achieved by using a generator expression (str(x) for x in data.iloc[row, 0:25]) to convert each value to a string and then joining them with a space in between using ' '.join(...).

The resulting string is then appended to the headlines list using the .append() method.

After the loop finishes, the headlines list will contain the concatenated strings of values from columns 0 to 24 for each row in the data DataFrame. This can be useful if you want to store or further process the concatenated headlines for later analysis or modeling tasks.

Note that the resulting headlines list will have the same length as the number of rows in the data DataFrame, and each element in the list represents the concatenated headline for the corresponding row.

In [41]:
headlines[0]

'a  hindrance to operations   extracts from the leaked reports scorecard hughes  instant hit buoys blues jack gets his skates on at ice cold alex chaos as maracana builds up for united depleted leicester prevail as elliott spoils everton s party hungry spurs sense rich pickings gunners so wide of an easy target derby raise a glass to strupar s debut double southgate strikes  leeds pay the penalty hammers hand robson a youthful lesson saints party like it s      wear wolves have turned into lambs stump mike catches testy gough s taunt langer escapes to hit     flintoff injury piles on woe for england hunters threaten jospin with new battle of the somme kohl s successor drawn into scandal the difference between men and women sara denver  nurse turned solicitor diana s landmine crusade put tories in a panic yeltsin s resignation caught opposition flat footed russian roulette sold out recovering a title'

In [17]:
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.ensemble import RandomForestClassifier

The code snippet y imports two modules from the sklearn library: CountVectorizer and RandomForestClassifier.CountVectorizer is a class in the sklearn.feature_extraction.text module. It is used for converting a collection of text documents into a matrix of token counts. It preprocesses the text data by tokenizing it into individual words or n-grams, and then counts the occurrences of these tokens in each document. This matrix representation can be used as input for various machine learning algorithms.

By importing CountVectorizer, you gain access to the functionality provided by this class, such as fitting a vectorizer on a corpus of text data, transforming text into a numerical representation, and extracting vocabulary information. It is commonly used for text classification, sentiment analysis, and other natural language processing tasks.RandomForestClassifier is a class in the sklearn.ensemble module. It is an implementation of the random forest algorithm for classification tasks. Random forests are an ensemble learning method that combines multiple decision tree classifiers and aggregates their predictions to make a final classification decision.

By importing RandomForestClassifier, you can create an instance of this class and utilize its methods for training a random forest classifier on labeled data, making predictions on new data, and evaluating the model's performance. Random forests are known for their ability to handle complex relationships in the data, handle missing values, and reduce overfitting.

Once these modules are imported, you can use the functionality provided by CountVectorizer and RandomForestClassifier to preprocess text data and train a random forest classifier on that data, respectively.

Please note that the code snippet you provided only imports these modules, and further code is required to utilize them effectively for specific tasks, such as preparing the data, splitting it into training and testing sets, fitting the classifier, making predictions, and evaluating the model's performance.

CountVectorizer is like a special tool that can help us understand and work with words in a bunch of texts. Imagine we have a lot of books or stories. CountVectorizer helps us convert those stories into a special chart that shows us how many times each word appears in each story.

First, CountVectorizer looks at all the words in the stories and counts how many times each word appears. Then, it creates a chart where each row represents a story, and each column represents a word. The numbers in the chart show how many times each word appears in each story.

This chart can be really helpful for machines to understand and analyze the stories. For example, we can use this chart to teach a machine how to recognize if a story is happy or sad based on the words used. Or we can use it to see which words are most important in the stories.

By importing CountVectorizer, we can use this special tool in our programs and tell the machine to do all these cool things with words! It helps us with things like understanding what the stories are about, figuring out people's feelings in the stories, and many other interesting things we can do with words.

CountVectorizer is like a friend that helps us understand words and stories better, so we can do fun and smart things with them!

RandomForestClassifier is like a special helper that can make decisions by looking at a lot of examples. Imagine we have a bunch of pictures of animals, and we want the computer to learn how to tell which animal is in each picture. RandomForestClassifier can help us with that!

First, RandomForestClassifier gathers a group of special friends called decision trees. Each decision tree is like a smart friend that can make decisions based on certain features of the pictures. They look at things like the color, shape, and size of the animals in the pictures.

Then, all the decision trees work together and share their opinions about which animal they think is in each picture. RandomForestClassifier listens to all the trees and combines their opinions to make a final decision about the animal in the picture.

This can be really helpful because each decision tree has its own way of looking at the pictures. By listening to all the trees, RandomForestClassifier can make more accurate decisions and understand the pictures better.

By importing RandomForestClassifier, we can use this special helper in our computer programs. We can show it lots of examples of animals and tell it which animal is in each picture. Then, RandomForestClassifier can learn from those examples and help us figure out which animal is in new pictures that it hasn't seen before.

RandomForestClassifier is like a group of friends that work together to help the computer understand and recognize different animals in pictures. It's like having a team of smart friends who know a lot about animals and can make good guesses about which animal is in each picture.

In [33]:
## implement BAG OF WORDS
countvector=CountVectorizer(ngram_range=(2,2))
traindataset=countvector.fit_transform(headlines)

his code implements the Bag-of-Words model using the CountVectorizer class from the scikit-learn library. Let's break it down step by step:

countvector = CountVectorizer(ngram_range=(2,2)): This line initializes an instance of the CountVectorizer class. The ngram_range=(2,2) parameter specifies that we want to consider only 2-word combinations, also known as bigrams. This means that the text will be tokenized into pairs of consecutive words.

traindataset = countvector.fit_transform(headlines): This line applies the CountVectorizer to a dataset of headlines, represented by the headlines variable. The fit_transform() method of the CountVectorizer class performs two operations:

fit(): This method learns the vocabulary from the headlines. It analyzes the text data and builds a dictionary of unique words (or bigrams in this case) called the vocabulary.
transform(): This method converts the headlines into a matrix representation, where each row corresponds to a headline and each column represents the count of a specific word or bigram in that headline.
The result, traindataset, is a sparse matrix that represents the bag-of-words representation of the headlines. Each row in the matrix corresponds to a headline, and each column represents a unique bigram from the vocabulary. The values in the matrix indicate the frequency of occurrence of each bigram in the respective headline.

To summarize, this code applies the Bag-of-Words model to a dataset of headlines using the CountVectorizer class. It tokenizes the text into bigrams and creates a matrix representation where each row represents a headline, each column represents a bigram, and the values indicate the frequency of each bigram in the respective headline.

In [34]:
# implement RandomForest Classifier
randomclassifier=RandomForestClassifier(n_estimators=200,criterion='entropy')
randomclassifier.fit(traindataset,train['Label'])

RandomForestClassifier(bootstrap=True, class_weight=None, criterion='entropy',
            max_depth=None, max_features='auto', max_leaf_nodes=None,
            min_impurity_decrease=0.0, min_impurity_split=None,
            min_samples_leaf=1, min_samples_split=2,
            min_weight_fraction_leaf=0.0, n_estimators=200, n_jobs=None,
            oob_score=False, random_state=None, verbose=0,
            warm_start=False)

In this code, we're using a special tool called "CountVectorizer" to help us count and understand words in some headlines. Imagine we have a list of headlines like "I love ice cream" or "The sun is shining." We want to see how many times certain pairs of words appear in these headlines.

First, we create a special box called "countvector." It's like a magic box that can help us count words in the headlines. We tell the box that we want to look at pairs of words, like "I love" or "the sun."

Then, we take our list of headlines and put them inside the magic box. The box starts counting how many times each pair of words appears in each headline. It's like having a little helper who counts for us!

After counting, the box gives us a special chart. In this chart, each row represents a headline, and each column represents a pair of words. The numbers in the chart show us how many times each pair of words appeared in each headline.

This chart is really helpful because we can use it to learn more about the headlines. We can see which pairs of words are used a lot and which are used less often. It helps us understand how the words are connected in the headlines.

So, by using CountVectorizer, we can count and understand the pairs of words in the headlines. It's like having a special box that helps us learn more about the words and how they are used together.

Here, we create an instance of the CountVectorizer class and assign it to the variable countvector. The CountVectorizer class helps us convert text data into a numerical representation that we can use for analysis or machine learning.

We set the parameter ngram_range to (2, 2), which means we want to consider pairs of words (bigrams) as features. This means that instead of looking at individual words, we look at two words together as a pair. For example, if we have a sentence like "I love ice cream," the bigrams would be "I love" and "love ice" and "ice cream."In this line, we use the fit_transform() method of the CountVectorizer instance (countvector) to convert our list of headlines (stored in the variable headlines) into a matrix of token counts.

The fit_transform() method does two things. First, it "fits" the CountVectorizer to our data, which means it learns the vocabulary of all the unique words and bigrams present in the headlines. Then, it "transforms" the headlines into a numerical matrix representation.

The resulting traindataset is a matrix where each row corresponds to a headline, and each column represents a unique word or bigram. The values in the matrix indicate the number of times each word or bigram appears in each headline. This matrix can be used as input for machine learning models or other analysis tasks.

So, in summary, this code helps us convert a list of headlines into a matrix of word and bigram counts, which can be used for further analysis or machine learning tasks.

In [35]:
## Predict for the Test Dataset
test_transform= []
for row in range(0,len(test.index)):
    test_transform.append(' '.join(str(x) for x in test.iloc[row,2:27]))
test_dataset = countvector.transform(test_transform)
predictions = randomclassifier.predict(test_dataset)

The code snippet you provided is used to predict the labels for the test dataset using a trained model. Let's break it down:In this block of code, we are preparing the test dataset for prediction. We create an empty list called test_transform to store the transformed version of each test example.

For each row in the test dataset, we iterate through the range of indices from 0 to the length of the test dataset. Then, we use the iloc method to select the columns from index 2 to 26 (columns 2 to 26) in that row of the test dataset.

Next, we convert the selected values to strings and join them together using a space as a separator. This creates a string representation of the features in that particular test example. Finally, we append the transformed string to the test_transform list.Here, we use the transform() method of the countvector object (which is an instance of CountVectorizer) to transform the test_transform data into a matrix representation similar to the one obtained during training. This ensures that the test data is represented in the same way as the training data.Finally, we use the predict() method of the randomclassifier object to make predictions on the transformed test dataset (test_dataset). The predict() method applies the trained model to the test dataset and returns the predicted labels for each example.

The resulting predictions variable holds the predicted labels for the test dataset based on the trained model. These predictions can be further analyzed or compared with the actual labels to evaluate the model's performance.

In summary, this code takes the test dataset, transforms it using the same CountVectorizer object used during training, and then makes predictions using a trained classifier (randomclassifier).

In [36]:
## Import library to check accuracy
from sklearn.metrics import classification_report,confusion_matrix,accuracy_score

The code  imports three modules from the sklearn.metrics library: classification_report, confusion_matrix, and accuracy_score. Let's discuss each of them:classification_report: This module provides a summary report of the main classification metrics. It computes various metrics such as precision, recall, F1-score, and support for each class in a classification problem. It is useful for evaluating the performance of a classification model and understanding the model's accuracy, precision, and recall for each class.

confusion_matrix: This module allows us to create a confusion matrix, which is a table that shows the performance of a classification model by summarizing the counts of true positive, true negative, false positive, and false negative predictions. It helps in visualizing the model's performance and understanding how well it is predicting each class.

accuracy_score: This module provides a function to calculate the accuracy score, which is a commonly used metric to evaluate the performance of a classification model. The accuracy score calculates the proportion of correctly predicted labels to the total number of samples. It gives us a single value that represents the overall accuracy of the model.

By importing these modules, you gain access to the functionality provided by each module. You can use classification_report to generate a summary report of classification metrics, confusion_matrix to create a confusion matrix, and accuracy_score to calculate the accuracy score.

These modules are commonly used for evaluating the performance of classification models and understanding how well the model is predicting the correct labels.

Please note that the code snippet you provided only imports these modules, and further code is required to utilize them effectively, such as providing the actual and predicted labels as inputs to calculate the classification metrics, confusion matrix, and accuracy score.

In [37]:
matrix=confusion_matrix(test['Label'],predictions)
print(matrix)
score=accuracy_score(test['Label'],predictions)
print(score)
report=classification_report(test['Label'],predictions)
print(report)

[[139  47]

 [ 13 179]]

0.8412698412698413

              precision    recall  f1-score   support



           0       0.91      0.75      0.82       186

           1       0.79      0.93      0.86       192



   micro avg       0.84      0.84      0.84       378

   macro avg       0.85      0.84      0.84       378

weighted avg       0.85      0.84      0.84       378




The code imported modules to evaluate the performance of the predicted labels (predictions) compared to the actual labels (test['Label']). Here, we use the confusion_matrix() function to calculate the confusion matrix. The confusion matrix gives us a table that shows the counts of true positives, true negatives, false positives, and false negatives. It helps us understand how well the model's predictions match the actual labels. The test['Label'] contains the actual labels, and predictions contains the predicted labels. The resulting matrix is stored in the matrix variable, and it is printed to the console using print(matrix).Next, we use the accuracy_score() function to calculate the accuracy score. The accuracy score measures the proportion of correctly predicted labels to the total number of samples. It gives us a single value that represents the overall accuracy of the model. The test['Label'] contains the actual labels, and predictions contains the predicted labels. The resulting accuracy score is stored in the score variable, and it is printed to the console using print(score).Finally, we use the classification_report() function to generate a report of classification metrics. This report includes metrics such as precision, recall, F1-score, and support for each class in the classification problem. It provides a summary of how well the model performed for each class. The test['Label'] contains the actual labels, and predictions contains the predicted labels. The resulting report is stored in the report variable, and it is printed to the console using print(report).

In summary, this code evaluates the performance of the predicted labels by comparing them to the actual labels using the confusion matrix, accuracy score, and classification report. It helps us understand how well the model is performing and provides insights into its precision, recall, F1-score, and overall accuracy.

This is the confusion matrix. It is a table that shows the performance of a classification model by summarizing the counts of true positives (top-left), true negatives (bottom-right), false positives (top-right), and false negatives (bottom-left). In this case, the matrix indicates that there are 139 true positives (correctly predicted positive labels), 47 false positives (incorrectly predicted positive labels), 13 false negatives (incorrectly predicted negative labels), and 179 true negatives (correctly predicted negative labels).This is the accuracy score. The accuracy score measures the proportion of correctly predicted labels to the total number of samples. In this case, the accuracy score is approximately 0.8413, which means that the model correctly predicted about 84.13% of the labels in the test dataset.This is the classification report. It provides various metrics such as precision, recall, and F1-score for each class (0 and 1). Precision represents the ability of the model to correctly identify positive samples, while recall represents the ability to find all positive samples. F1-score is a combination of precision and recall, providing a balanced measure of the model's performance. The "support" column indicates the number of samples for each class.

In this case, for class 0, the precision is 0.91 (91%), which means that out of all the predicted positive labels, 91% were actually positive. The recall is 0.75 (75%), indicating that the model correctly identified 75% of the positive samples. The F1-score is 0.82 (82%), which is a balanced measure of precision and recall. There are 186 samples in class 0.

For class 1, the precision is 0.79 (79%), meaning that out of all the predicted negative labels, 79% were actually negative. The recall is 0.93 (93%), indicating that the model correctly identified 93% of the negative samples. The F1-score is 0.86 (86%), providing a balanced measure. There are 192 samples in class 1.

Overall, the classification report gives a summary of the model's performance for each class, indicating its precision, recall, and F1-score.